File: RSNPUnitConnector_Socket_sample/Java/src/main/SocketSample.java
package main;

import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.PrintWriter;
import java.net.Socket;
import java.util.Properties;


public class SocketSample {
	public static void main(String[] args) throws IOException {

		String ipaddress;
		String port;
		int count;

		//read parameter
		Properties properties = new Properties();
		String pass_str = "Config.properties";
		InputStream istream = new FileInputStream(pass_str);
		properties.load(istream);

		ipaddress = properties.getProperty("IPaddress");
		port = properties.getProperty("Port");
		int port_int = Integer.valueOf(port).intValue();

		try {
			//make socket
			Socket csocket = new Socket(ipaddress, port_int);
			System.out.print("IPaddress: "+ipaddress);
			System.out.print("Port: "+ port);
			count = 0;

			PrintWriter writer = new PrintWriter(csocket.getOutputStream(), true);

			while (true) {
				count += 1;
				String sample_data= "{\"data\": [{\"ac_id\": \"1\", \"ac\": \"robot_state\", \"re_id\": \"1\", \"re\": "+count+", \"co\": \"\"}]}";

				writer.println(sample_data);
				System.out.println(sample_data);

				try {
					Thread.sleep(5000);
				} catch (InterruptedException e) {
				}

		        }
			} catch (Exception e) {
		      e.printStackTrace();
		    }


	}

}

File: RSNPUnitConnector_Socket_sample/Python/Socket_sample.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# -*- Python -*-

import socket
import time
import ConfigParser #for Python2
#import configparser #for Python3

def main():

    # setting of configparser object
    config = ConfigParser.ConfigParser() #for Python2
    # config = configparser.ConfigParser() #for Python3

    # read Config.ini
    config.read("./Config.ini")

    # reading parameter of Config.ini
    IPaddress = config.get('Settings','IPaddress')
    Port = config.get('Settings','Port')
    Port = int(Port)
    print("IPaddr = [{}] Port = [{}]".format(IPaddress, Port))

    # setting of socket object
    Socket_client = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

    # setting of IPaddress and Port
    Socket_client.connect((IPaddress, Port))

    # inicialize result data
    result_data = '"hoge"'
    count = 0
    # setting of json format data to send RSNPUnit
    #send_data = '{"data": [{"ac_id": "1", "ac": "robot_state", "re_id": "1", "re": '+result_data+', "co": ""},{"ac_id": "2", "ac": "robot_state", "re_id": "2", "re": '+result_data+', "co": ""}]}'

    while True:

        time.sleep(180)  #5s interval

        count += 1

        send_data = '{"data": [{"ac_id": "1", "ac": "robot_state", "re_id": "1", "re": '+result_data+', "co": ""},{"ac_id": "2", "ac": "robot_state", "re_id": "2", "re": '+str(count)+', "co": ""}]}'

        # send data to RSNPUnit by Socket communication
        Socket_client.send(send_data)

        #only print data
        print("send_data: " + send_data)


if __name__ == "__main__":
    main()